### ISSUE TITLE
Code Lab Feedback

### ISSUE THREAD

**Issue by zacharyweston** [zacharyweston/codelab-sinatra-helper-methods#1]
Hey @chrisvfritz, can you take a look at this? It's [hosted here](https://guarded-castle-45814.herokuapp.com/) and meets the following [project](http://lansingcodelab.com/lessons/sinatra-helper-methods/1) criteria:
- [x] Define and use a method called `pluralize` in your app. It should take two parameters: `singular_name` and `count`. `pluralize('cat', 5)` should return `'5 cats'`, `pluralize('cat', 1)` should return `'1 cat'`, and `pluralize('cat', 0)` should return `'0 cats'`.
- [x] Define and use a method called `use_google_font` in your app. It should take a single parameter: `font`. It should return the `link` element to include the font and a `style` element that applies the font to the body. Example usage would be `use_google_font('Open Sans')`. This one requires you know how to [customize the fonts in a webpage](http://lansingcodelab.com/lessons/css-font-families/1), so complete or review that lesson if you need to.
- [x] Define and use a method called `icon_button` in your app. It should take 3 parameters: `icon`, `type` (with a default value of `'default'`), and `message`, with a default value of `''`. That means the only required parameter is `icon`. It should return a button with an icon, similar to the [example in the Bootstrap docs](http://getbootstrap.com/components/#glyphicons-examples). Note that Bootstrap must be correctly included in your app for this to work.

**Comment by chrisvfritz** [comment]
This is looking very good overall, but I do see a few issues:
1. You're [not _quite_ serving valid HTML](https://validator.w3.org/check?uri=https%3A%2F%2Fguarded-castle-45814.herokuapp.com%2F&charset=%28detect+automatically%29&doctype=Inline&group=0) here.
2. There are a few inconsistent indentations. While almost everything is indented 2 spaces, there are some cases where I see [4-space indentation](https://github.com/zacharyweston/codelab-sinatra-helper-methods/blob/master/app.rb#L27-L29) or even [no indentation](https://github.com/zacharyweston/codelab-sinatra-helper-methods/blob/master/app.rb#L46) where there should be.
3. This ties into the first issue, but once you're serving valid HTML, it'd probably be best to just include Bootstrap once in the head of the page, rather than with every button. You'd only use helpers like this if your site were _already_ using Bootstrap.
4. I don't see the correct defaults set for the `icon_button` method.

Please don't hesitate if you have any questions about these (or just in general). :smiley:

**Comment by zacharyweston** [comment]
Makes sense, new code has been pushed to GitHub, and the hosted site is the same

**Comment by chrisvfritz** [comment]
1. I'm still seeing [some validation errors](https://validator.w3.org/nu/?doc=https%3A%2F%2Fguarded-castle-45814.herokuapp.com%2F). :cry: 
2. Indentation looks great now! :star: 
3. Nice Bootstrap inclusion in the `head`! :star: 
4. Excellent-looking defaults! :star: 

Let me know if you need any help working out those HTML validation errors. While you're still getting used to working in Ruby projects, it's definitely definitely a good idea to take advantage of [validator.w3.org](https://validator.w3.org/) to make sure your code is indeed creating valid HTML!

**Comment by zacharyweston** [comment]
New code has been pushed, new URL https://evening-castle-45690.herokuapp.com/

**Comment by chrisvfritz** [comment]
:rocket: :shipit: :smiley:
